File: AI_Crop_Disease_Detection_System/AI Frontend/lib/src/models/crop.dart
import 'disease.dart';

class Crop {
  const Crop({
    required this.id,
    required this.name,
    String? canonicalName,
    this.scientificName,
    this.description,
    this.diseases = const [],
  }) : canonicalName = canonicalName ?? name;

  final int id;
  final String name;
  final String canonicalName;
  final String? scientificName;
  final String? description;
  final List<Disease> diseases;

  factory Crop.fromJson(Map<String, dynamic> json) {
    final diseasesJson = json['diseases'] as List<dynamic>? ?? [];

    return Crop(
      id: json['id'] as int,
      name: json['name'] as String,
      canonicalName: json['canonical_name'] as String?,
      scientificName: json['scientific_name'] as String?,
      description: json['description'] as String?,
      diseases: diseasesJson
          .map((disease) => Disease.fromJson(disease as Map<String, dynamic>))
          .toList(),
    );
  }
}

File: AI_Crop_Disease_Detection_System/AI Backend/app/Models/Crop.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Crop extends Model
{
    protected $fillable = [
        'name',
        'name_sn',
        'scientific_name',
        'description',
        'description_sn',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
        ];
    }

    public function diseases(): HasMany
    {
        return $this->hasMany(Disease::class);
    }
}
